lib/task/tools/deleteDescriptionTask.class.php

```diff
@@ -49,44 +53,115 @@ protected function configure()
    */
   public function execute($arguments = array(), $options = array())
   {
-    $databaseManager = new sfDatabaseManager($this->configuration);
-    $conn = $databaseManager->getDatabase('propel')->getConnection();
-    sfContext::createInstance($this->configuration);
+    parent::execute($arguments, $options);
+
+    $this->resourceType = $options['repository'] ? 'QubitRepository' : 'QubitInformationObject';
+    $this->fetchResource($arguments['slug']);
+
+    if (!$this->confirmDeletion($options['no-confirmation']))
+    {
+      $this->logSection(sprintf('[%s] Task aborted.', strftime('%r')));
+      return;
+    }
+
+    // User wishes to proceed, delete targeted information objects.
+    switch ($this->resourceType)
+    {
+      case 'QubitRepository':
+        $this->deleteDescriptionsFromRepository();
+        break;
+      case 'QubitInformationObject':
+        $this->deleteDescriptions($this->resource);
+        break;
+    }
+
+    $this->logSection(sprintf('[%s] Finished: %d descriptions deleted.', strftime('%r'), $this->nDeleted));
+  }
+
+  /**
+   * Allow the user to bail out if they aren't sure they want to delete targeted descriptions.
+   *
+   * @param $noConfirmation  Whether or not to bypass the confirmation warning (true = bypass).
+   * @return bool  True if we want to proceed with the task, false if we want to abort.
+   */
+  private function confirmDeletion($noConfirmation)
+  {
+    switch ($this->resourceType)
+    {
+      case 'QubitRepository':
+        $confirmWarning = sprintf('WARNING: You are about to delete all the records under the repository "%s".',
+                                  $this->resource->getAuthorizedFormOfName(array('cultureFallback' => true)));
+        break;
+      case 'QubitInformationObject':
+        $confirmWarning = sprintf('WARNING: You are about to delete the record "%s" and %d descendant records.',
+                                  $this->resource->getTitle(array('cultureFallback' => true)),
+                                  max(0, count($this->resource->descendants->andSelf()) - 1));
+        break;
+    }
 
-    if (null === $informationObject = QubitInformationObject::getBySlug($arguments['slug']))
+    if ($noConfirmation || $this->askConfirmation(array($confirmWarning,
```

Comment:
Not a big deal, but checking `$noConfirmation` first could avoid counting descendants in the previous switch